### DISCUSSION TITLE
[Question] Edited: How to run IOS 15 and 16

### DISCUSSION THREAD

**Discussion by Autumnlight02** [mozilla/pdf.js#19076] [General]
I wanna use this on ios, but simd seems to be broken up until ios 16, any chance of using this library without simd?

Edit: read further below for solution to issue in title:

**Comment by calixteman** [comment]
What do you mean by `simd` and why do you think pdf.js uses `simd` ?

**Reply by Autumnlight02** [reply]
@calixteman Sorry solved, in the team I am working in we had a simd check for this library, seems like one of our developers mistakenly made a check if this library works by checking for simd, even though they're not related. it just happen to work.

For our team I will patch now the code, since IOS 13 14 15 cannot handle the static { ... syntax which appears in a file. Would this change be wished to be as a commit or should we keep that change local? it just happens once in Viewer.mjs and my assumption is if we replace it we should have older device compability going. But thank you that you took your time to respond to my question.

**Reply by Autumnlight02** [reply]
Inside of the legacy viewer: 
![image](https://github.com/user-attachments/assets/059f1e0f-d17d-4518-b868-0a6e6a9d8732)

**Comment by calixteman** [comment]
Static blocks should be supported on Safari 16.4 for iOS:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Static_initialization_blocks#browser_compatibility

and our legacy build is supposed to be fine with this version of Safari:
https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#legacy-build

If your version of Safari isn't recent enough, you'll have to patch that stuff by yourself.

**Reply by Autumnlight02** [reply]
@calixteman Thank you for the info, may I do PR requests to support older IOS versions as well or is it not wished to have such fixes in the codebase? I suspect that I have to change about 100 lines max.

**Comment by calixteman** [comment]
We won't accept any PRs to support too old versions (too much extra work, maintenance, ...)
That said, maybe you could just hack on `gulpfile.js` in order to build something compatible with your environment:
https://babeljs.io/docs/babel-plugin-transform-class-static-block

**Reply by Autumnlight02** [reply]
Did work out, thank you for your help!

**Reply by Autumnlight02** [reply]
@calixteman just a last short message: By applying your recommended patch it seems to flawlessly work down to ios version 15 which supports all devices after iphone +6s. If someone else asks for this you can just refer them to your message. 

Thank you again for your help and I wish you an amazing day!
